src/main/java/org/codehaus/mojo/servicedocgen/ServiceDocGenReport.java

```diff
@@ -176,23 +186,35 @@ protected void executeReport( Locale locale )
                 return;
             }
             Analyzer analyzer = new Analyzer( getLog(), this.project, getProjectClassloader(), builder, this.descriptor,
-                                              this.introspectFields );
+                this.introspectFields );
             ServicesDescriptor services = analyzer.createServicesDescriptor( serviceClasses );
 
-            getLog().info( "Generating output..." );
-            ServicesGenerator generator =
-                new VelocityServicesGenerator( Util.appendPath( this.templatePath, this.templateName ) );
-
-            File reportDirectory = new File( this.outputDirectory, this.reportFolder );
-            if ( !reportDirectory.isDirectory() )
+            String[] reportTypes = this.reportType.split(",");
+            String[] templateNames = this.templateName.split(",");
+            String[] outputNames = this.outputName.split(",");
+            for( int i = 0; i < reportTypes.length; i++ )
             {
-                boolean ok = reportDirectory.mkdirs();
-                if ( !ok )
+
+                if ( reportTypes[i].trim().equals(ReportType.OPENAPI_JSON.toString()) || reportTypes[i].trim().equals(ReportType.OPENAPI_YAML.toString()) )
+                {
+                    sortServiceOperationsByPath(services);
+                }
```

Comment:
why is this specific for OpenAPI?
why should this method be called multiple times in the loop? Calling it a second time will not cause any change...
For simplicity I would always do it once before looping or never do it.

Reply:
I moved it over the loop.
I had to sort the services by the name of the path, because in the OpenApi definition I cannot create a separate 'section' for each operation with the same path. So I needed operations with the same path directly next to each other.
```
paths:
  /demo:
    get:

  /demo:
    post:
```
This would not be valid. I have to do it this way:
```
paths:
  /demo:
    get:
    post:
```